src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -137,12 +132,15 @@ object Task {
 
     // Regular expression for matching taskIds since instance-era
     private val TaskIdWithInstanceIdRegex = """^(.+)\.(instance-|marathon-)([^_\.]+)[\._]([^_\.]+)$""".r
+    private val ResidentTaskIdWithInstanceIdRegex = """^(.+)\.(instance-|marathon-)([^_\.]+)[\._]([^_\.]+)\.(\d+)$""".r
 
     private val uuidGenerator = Generators.timeBasedGenerator(EthernetAddress.fromInterface())
 
     def runSpecId(taskId: String): PathId = {
       taskId match {
-        case TaskIdWithInstanceIdRegex(runSpecId, prefix, instanceId, maybeContainer) => PathId.fromSafePath(runSpecId)
+        case ResidentTaskIdWithInstanceIdRegex(runSpecId, prefix, uuid, container, attempt) =>
```

Comment:
`(runSpecId, _, _, _, _)`?

Reply:
It is less-readable. I intentionally kept it the way it was and still is.

Reply:
Doesn't scala (or linter) complain about not used parameters?

Reply:
Scala doesn't complain. By linter you mean Scapegoat? If yes, then it is fine too.